src/Bicep.LangServer/Handlers/BicepGenerateParamsCommandHandler.cs

```diff
@@ -0,0 +1,147 @@
+
+            // If the template exists and contains bicep generator metadata, we can go ahead and replace the file.
+            // If not, we'll fail the generate params.
+            if (File.Exists(compiledFilePath) && !TemplateContainsBicepGeneratorMetadata(File.ReadAllText(compiledFilePath)))
+            {
+                return "Generate Params failed. The file \"" + compiledFile + "\" already exists and was not generated by Bicep. If overwriting the file is intended, delete it manually and retry the Generate Params command.";
+            }
+
+            var fileUri = documentUri.ToUri();
+            RootConfiguration? configuration = null;
+
+            try
+            {
+                configuration = this.configurationManager.GetConfiguration(fileUri);
+            }
+            catch (ConfigurationException exception)
+            {
+                // Fail the generate params if there's configuration errors.
+                return exception.Message;
+            }
+
+            CompilationContext? context = compilationManager.GetCompilation(fileUri);
+            Compilation compilation;
+
+            if (context is null)
+            {
+                SourceFileGrouping sourceFileGrouping = SourceFileGroupingBuilder.Build(this.fileResolver, this.moduleDispatcher, new Workspace(), fileUri, configuration);
+                compilation = new Compilation(features, namespaceProvider, sourceFileGrouping, configuration, new LinterAnalyzer(configuration));
+            }
+            else
+            {
+                compilation = context.Compilation;
+            }
+
+            KeyValuePair<BicepFile, IEnumerable<IDiagnostic>> diagnosticsByFile = compilation.GetAllDiagnosticsByBicepFile()
+                .FirstOrDefault(x => x.Key.FileUri == fileUri);
+
+            if (diagnosticsByFile.Value.Any(x => x.Level == DiagnosticLevel.Error))
+            {
+                return "Generate Params failed. Please fix below errors:\n" + DiagnosticsHelper.GetDiagnosticsMessage(diagnosticsByFile);
+            }
+
+            using var fileStream = new FileStream(compiledFilePath, FileMode.Create, FileAccess.ReadWrite);
+            var emitter = new TemplateEmitter(compilation.GetEntrypointSemanticModel(), emitterSettings);
+            EmitResult result = emitter.EmitParametersFile(fileStream);
+
+            return "Generate Params succeeded. Created file " + compiledFile;
+        }
+
+        // Returns true if the template contains bicep _generator metadata, false otherwise
+        public bool TemplateContainsBicepGeneratorMetadata(string template)
+        {
+            try
+            {
+                if (!string.IsNullOrEmpty(template))
+                {
+                    JToken jtoken = template.FromJson<JToken>();
+                    if (TemplateHelpers.TryGetTemplateGeneratorObject(jtoken, out DeploymentTemplateGeneratorMetadata generator))
```

Comment:
>TryGetTemplateGeneratorObject [](http://example.com/codeflow?start=40&length=29)

Do param files actually allow open-ended `metadata` like templates do? I don't see it documented here [https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/parameter-files](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/parameter-files), but I could be missing something?

Reply:
I tried to deploy a generated arm template with `metadata` structure in the parameters.json file, it went successfully and deployed.

I decided to keep the `metadata` structure to have some information in the parameters file too.

if you see it's not a good fit, I can remove it.

Reply:
@polatengin  I would like for "metadata" to be removed. It doesn't fit into our best practices